internal/commands/completion.go

```diff
@@ -0,0 +1,131 @@
+package commands
+
+import (
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	completionLongDesc = `
+Output shell completion code for bash.
+The shell code must be evaluated to provide interactive
+completion of qbec commands.  This can be done by sourcing it from
+the .bash_profile.`
+
+	completionExample = `
+# If running bash-completion, write the output code to your
+bash_completion.d/ directory,
+	qbec completion > /usr/local/etc/bash_completion.d/qbec
+
+# To load the completion into your current shell
+	source <(qbec completion)
+
+# Save the completion script to a file and source it in your ~/.bash_profile
+	qbec completion > ~/qbec.bash
+	printf "
+	# qbec bash completion
+	source '~/qbec.bash'
+	" >> ~/.bash_profile
+	source ~/.bash_profile
+`
+	BashCompletionFunc = `
+__qbec_override_root() {
+    local root pair
+    for w in "${words[@]}"; do
+    	if [ -n "${pair}" ]; then
+	    root="--root=${w}"
+	fi
+        case "${w}" in
+            --root=*)
+		root="${w}"
+                ;;
+            --root)
+	    	pair=1
+                ;;
+        esac
+    done
+    if [ -n "${root}" ]; then
+    	echo -n "${root} "
+    fi
+}
+
+__qbec_override_env() {
+    local env="_"
+    if [[ ${#nouns[@]} -gt 0 ]]; then
+    	env="${nouns[${#nouns[@]} -1]}"
+    fi
+    echo -n " $env "
+}
+
+__qbec_get_components() {
+    if components_list=$(qbec $(__qbec_override_root) component list $(__qbec_override_env) -o json 2>/dev/null | jq -r .[].Name); then
```

Comment:
curious for your thoughts on whether relying on `jq` is safe enough for all users of qbec. 

alternatively i can use awk and parse the default output format.